cli/cmd/resource_groups.go

```diff
@@ -0,0 +1,291 @@
+			err := cli.LwApi.V2.ResourceGroups.Delete(args[0])
+			if err != nil {
+				return errors.Wrap(err, "unable to delete resource group")
+			}
+			return nil
+		},
+	}
+
+	// create command is used to create a new lacework resource group
+	resourceGroupsCreateCommand = &cobra.Command{
+		Use:   "create",
+		Short: "create a new resource group",
+		Long:  "Creates a new single resource group.",
+		RunE: func(_ *cobra.Command, args []string) error {
+			if !cli.InteractiveMode() {
+				return errors.New("interactive mode is disabled")
+			}
+
+			err := promptCreateResourceGroup()
+			if err != nil {
+				return errors.Wrap(err, "unable to create resource group")
+			}
+
+			cli.OutputHuman("The resource group was created.\n")
+			return nil
+		},
+	}
+)
+
+func promptCreateResourceGroup() error {
+	var (
+		group  = ""
+		prompt = &survey.Select{
+			Message: "Choose a resource group type to create: ",
+			Options: []string{
+				"Aws",
+				"Azure",
+				"Container",
+				"Gcp",
+				"LwAccount",
+				"Machine",
+			},
+		}
+		err = survey.AskOne(prompt, &group)
+	)
+	if err != nil {
+		return err
+	}
+
+	switch group {
+	case "Aws":
+		return createAwsResourceGroup()
+	case "Azure":
+		return createAzureResourceGroup()
+	case "Container":
+		return createContainerResourceGroup()
+	case "Gcp":
+		return createGcpResourceGroup()
+	case "LwAccount":
+		return createLwAccountResourceGroup()
+	case "Machine":
+		return createMachineResourceGroup()
+	default:
+		return errors.New("unknown resource group type")
+	}
+}
+
+func buildResourceGroupPropsTable(group resourceGroup) string {
+	props := determineResourceGroupProps(group.ResType, group.Props)
+
+	return renderOneLineCustomTable("RESOURCE GROUP PROPS",
+		renderCustomTable([]string{}, props,
+			tableFunc(func(t *tablewriter.Table) {
+				t.SetBorder(false)
+				t.SetColumnSeparator(" ")
+				t.SetAutoWrapText(false)
+				t.SetAlignment(tablewriter.ALIGN_LEFT)
+			}),
+		),
+		tableFunc(func(t *tablewriter.Table) {
+			t.SetBorder(false)
+			t.SetAutoWrapText(false)
+		}),
+	)
+}
+
+func determineResourceGroupProps(resType string, props interface{}) [][]string {
+	details := setBaseProps(props)
+
+	switch resType {
+	case api.AwsResourceGroup.String():
+		details = append(details, setAwsProps(props))
+	case api.AzureResourceGroup.String():
+		details = append(details, setAzureProps(props)...)
+	case api.ContainerResourceGroup.String():
+		details = append(details, setContainerProps(props)...)
+	case api.GcpResourceGroup.String():
+		details = append(details, setGcpProps(props)...)
+	case api.LwAccountResourceGroup.String():
+		details = append(details, setLwAccountProps(props))
+	case api.MachineResourceGroup.String():
+		details = append(details, setMachineProps(props))
+	}
+
+	return details
+}
+
+func setBaseProps(props interface{}) [][]string {
+	var (
+		baseProps resourceGroupPropsBase
+		details   [][]string
+	)
+
+	err := json.Unmarshal([]byte(props.(string)), &baseProps)
```

Comment:
we are doing this casting `props.(string)` often, could we do it just once at the beginning of the function `determineResourceGroupProps()` and pass a `string` instead of an `interface{}`